packages/front-end/src/app/components/EGPageHeader.vue

```diff
@@ -1,17 +1,36 @@
 <script setup lang="ts">
+  import { defineProps, computed } from 'vue';
+  import { useRouter } from 'vue-router';
+
   const props = withDefaults(
     defineProps<{
       title: string;
       description?: string;
-      showBack?: boolean;
+      showBack: boolean;
       backButtonLabel?: string;
-      backAction: () => void;
+      backAction?: () => void;
+      isLoading?: boolean;
+      skeletonConfig?: {
+        titleLines?: number;
+        descriptionLines?: number;
+      };
     }>(),
     {
+      backButtonLabel: 'Back',
+      isLoading: false,
+      skeletonConfig: {
+        titleLines: 1,
+        descriptionLines: 2,
+      },
+      backAction: () => {},
       description: '',
-      showBack: true,
     },
   );
+
+  const router = useRouter();
+
+  const skeletonTitleLines = computed(() => props.skeletonConfig?.titleLines ?? 1);
+  const skeletonDescriptionLines = computed(() => props.skeletonConfig?.descriptionLines ?? 1);
```

Comment:
is this a double up on defaults when there's already default values in the props definition?